### ISSUE TITLE
Build candidates still reporting old count

### ISSUE THREAD

**Issue by finestructure** [SwiftPackageIndex/SwiftPackageIndex-Server#909]
The new build candidate query selects 5 candidates as of this moment, while the old one selects 7.

Monitoring still reflects the old count:

<img width="496" alt="CleanShot 2021-02-01 at 09 20 57@2x" src="https://user-images.githubusercontent.com/65520/106432324-d5bc4e00-646e-11eb-975d-2ecf2ec9a2e3.png">

Need to investigate what drives monitoring (should be the same source) or why this isn't being picked up yet. 2.19.0 has been live for 25mins now, so it should all be reporting up-to-date figures.

**Comment by finestructure** [comment]
SQL query parameter interpolation doesn't seem to be working as expected. Can reproduce behaviour running with prod snapshot. This selects the "old" amount:

```
                        v.created_at < NOW() - INTERVAL '\(bind: Constants.branchBuildDeadTime) hours'
```

whereas this selects as expected:

```
                        v.created_at < NOW() - INTERVAL '24 hours'
```

We have explicit tests for the query string behaviour and I'm baffled why they would pass if this isn't working.

**Comment by finestructure** [comment]
I think I've got a minimal repro:

```
    func test_minimal() throws {
        let db = try XCTUnwrap(app.db as? SQLDatabase)
        struct Row: Codable { var id: Int }
        try db.raw("drop table if exists t").run().wait()
        try db.raw("create table t ( id int, created_at timestamp with time zone )").run().wait()

        do {
            try db.raw("insert into t values (1, NOW() - interval '1h')").run().wait()
            do { // hardcoded test
                let rows = try db.raw(
                    "select id from t where created_at >= now() - interval '2h'"
                ).all(decoding: Row.self).wait().map(\.id)
                XCTAssertEqual(rows, [1])
            }
            do { // bind test
                let rows = try db.raw(
                    "select id from t where created_at >= now() - interval '\(bind: 2)h'"
                ).all(decoding: Row.self).wait().map(\.id)
                XCTAssertEqual(rows, [1]) // FAILS - is []
            }
        }
    }
```

**Comment by finestructure** [comment]
The fix could be to use `\(raw: ...)` here.

```swift
        do { // bind test
            let rows = try db.raw(
                "select id from t where created_at >= now() - interval '\(raw: "2")h'"
            ).all(decoding: Row.self).wait().map(\.id)
            XCTAssertEqual(rows, [1]) // PASSES
        }
```

I _did_ have a concern when implementing the interval changes that interpolation into the quoted `'...'` interval parameter might cause issues but since neither the type system nor the tests complained I thought this was ok.

I need to go over our tests again but either we've managed to hit a set of parameters to make the tests pass or we've got a test gap.

**Comment by finestructure** [comment]
This does indeed fix it. I used the same patters in `trimBuilds`:

```swift
     return db.raw("""
        DELETE
        FROM builds b
        USING versions v
        WHERE b.version_id = v.id
        AND (
            v.latest is null
            OR
            (b.status = 'pending' AND b.created_at < NOW() - INTERVAL '\(bind: Constants.trimBuildsGracePeriod) hours')
        )
        RETURNING b.id
        """)
```

The implications are that we _overselected_ builds for deletion. Unless I'm mistaken, the bind parameter gets rendered as `0` according to my testing.

That means we over-aggressively purged pending builds from the db. This likely had little effect on operations. Here's why:

- we create pending records in the db
- we then trigger builds
- we delete the pending records. This does not prevent builds from reporting back - they'll just re-create builds. What does happen is that `pending` (hourglass symbols) get turned into question marks in the UI. That's why we were seeing more than expected I believe.
- worst case we'd enqueue a job again because we've no pending record
- in practise this likely didn't happen a lot, because we don't enqueue more jobs when we've got a full pipeline

We did quite likely process more builds than necessary though. We may see quite a significant reduction if that's the case (although it'll be hard to tell if it's the updated build trigger query or this change.)
